File: app/src/main/java/com/example/ela2/Submenu2.java
package com.example.ela2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class Submenu2 extends AppCompatActivity {
    private Button Submenu2_1;
    private Button Submenu2_2;
    private Button Submenu2_3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_submenu2);

        Submenu2_1 = findViewById(R.id.button2_1); //Click for the Submenu 1
        Submenu2_2 = findViewById(R.id.button2_2); //Click for the Submenu 2
        Submenu2_3 = findViewById(R.id.button2_3); //Click for the Submenu 3

        //Function for Navigation to Submenu1
        Submenu2_1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //Intent intent1 = new Intent(MainActivity.this,Submenu1.class);
                //startActivity(intent1); //intent1 stands for Submenu1
                startActivity(new Intent(Submenu2.this,Submenu2_1.class));
            }
        });

        //Function for Navigation to Submenu3
        Submenu2_2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //Intent intent3 = new Intent(MainActivity.this,Submenu3.class);
                //startActivity(intent3); //intent3 stands for Submenu3
                //OR
                startActivity(new Intent(Submenu2.this,Submenu2_2.class));
            }
        });

        //Function for Navigation to Submenu3
        Submenu2_3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //Intent intent3 = new Intent(MainActivity.this,Submenu3.class);
                //startActivity(intent3); //intent3 stands for Submenu3
                //OR
                startActivity(new Intent(Submenu2.this,Submenu2_3.class));
            }
        });
    }
}

File: app/src/main/java/com/example/ela2/Submenu2.kt
package com.example.ela2

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.example.ela2.databinding.ActivitySubmenu2Binding

class Submenu2 : Fragment() {
    private var _binding: ActivitySubmenu2Binding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = ActivitySubmenu2Binding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.button21.setOnClickListener {
            findNavController().navigate(R.id.action_submenu2Fragment_to_submenu21Fragment)
        }

        binding.button22.setOnClickListener {
            findNavController().navigate(R.id.action_submenu2Fragment_to_submenu22Fragment)
        }

        binding.button23.setOnClickListener {
            findNavController().navigate(R.id.action_submenu2Fragment_to_submenu23Fragment)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
